src/browser/BrowserService.cpp

```diff
@@ -0,0 +1,570 @@
+                        entry->endUpdate();
+                    }
+                }
+            }
+}
+
+QList<Entry*> BrowserService::searchEntries(Database* db, const QString& hostname)
+{
+    QList<Entry*> entries;
+    if (Group* rootGroup = db->rootGroup())
+        for (Entry* entry : EntrySearcher().search(hostname, rootGroup, Qt::CaseInsensitive)) {
+            QString title = entry->title();
+            QString url = entry->url();
+
+            //Filter to match hostname in Title and Url fields
+            if ((!title.isEmpty() && hostname.contains(title))
+                || (!url.isEmpty() && hostname.contains(url))
+                || (matchUrlScheme(title) && hostname.endsWith(QUrl(title).host()))
+                || (matchUrlScheme(url) && hostname.endsWith(QUrl(url).host())) ) {
+                entries.append(entry);
+            }
+        }
+    return entries;
+}
+
+QList<Entry*> BrowserService::searchEntries(const QString& text)
+{
+    //Get the list of databases to search
+    QList<Database*> databases;
+    if (BrowserSettings::searchInAllDatabases()) {
+        for (int i = 0; i < m_dbTabWidget->count(); i++)
+            if (DatabaseWidget* dbWidget = qobject_cast<DatabaseWidget*>(m_dbTabWidget->widget(i)))
+                if (Database* db = dbWidget->database())
+                    databases << db;
+    }
+    else if (DatabaseWidget* dbWidget = m_dbTabWidget->currentDatabaseWidget()) {
+        if (Database* db = dbWidget->database())
+            databases << db;
+    }
+
+    //Search entries matching the hostname
+    QString hostname = QUrl(text).host();
+    QList<Entry*> entries;
+    do {
+        for (Database* db : databases)
+            entries << searchEntries(db, hostname);
+    } while(entries.isEmpty() && removeFirstDomain(hostname));
+
+    return entries;
+}
+
+void BrowserService::removeSharedEncryptionKeys()
+{
+    if (!isDatabaseOpened()) {
+        QMessageBox::critical(0, tr("KeePassXC: Database locked!"),
+                              tr("The active database is locked!\n"
+                                 "Please unlock the selected database or choose another one which is unlocked."),
+                              QMessageBox::Ok);
+    } else if (Entry* entry = getConfigEntry()) {
+        QStringList keysToRemove;
+        for (const QString& key : entry->attributes()->keys())
+            if (key.startsWith(ASSOCIATE_KEY_PREFIX))
+                keysToRemove << key;
+
+        if(keysToRemove.count()) {
+            entry->beginUpdate();
+            for (const QString& key : keysToRemove)
+                entry->attributes()->remove(key);
+            entry->endUpdate();
+
+            const int count = keysToRemove.count();
+            QMessageBox::information(0, tr("KeePassXC: Removed keys from database"),
+                                     tr("Successfully removed %1 encryption-%2 from KeePassXC Settings.").arg(count).arg(count ? "keys" : "key"),
```

Comment:
This is not proper plural handling. Please see https://doc.qt.io/qt-5/i18n-source-translation.html#handling-plurals